### PR TITLE
[Fix] Fetch providers from registry

### LINKED ISSUE #445: Settings: Manage Liquidity Providers: Reload the list in redux after a provider is deleted or added
Now if someone delete a defaiult provider (ie. got from registry)  the only way to reload the list of providers is to kill and restart the app, but doing this will fetch and replace from registry at restart, making impossible to delete one basically.

### PR DESCRIPTION
**Pull request by louisinger**
This PR simplifies the flow of `restoreProviders` (the saga function launched each time the user "LOGIN").

now the list of TDEX providers is done following these steps:
1. try to "restore" providers persisted in local storage.
2. **if no providers in step (1)** fetch the list of providers from registry.
3. **if an error happens in step (2)**: add the default provider (depends on network type).

it closes #445 

@altafan please review

### CODE REVIEW

**Comment by louisinger** [comment]
Revert [this commit](https://github.com/tdex-network/tdex-app/pull/446/commits/2b8c684162433ea50cf75a7614f7826aeed89ca1) when  [tdex-sdk PR#118](https://github.com/tdex-network/tdex-sdk/pull/118) is merged

File: src/redux/sagas/tdexSaga.ts [hunk 1]
```diff
@@ -48,46 +50,38 @@ function* fetchMarkets({ payload }: { payload: TDEXProvider }) {
...
+    if (providersFromStorage.length > 0) {
+      return providersFromStorage;
     }
+  } catch {
+    yield put(addErrorToast(FailedToRestoreProvidersError));
+  }
+
+  // try to fetch providers from registry only on liquid
+  try {
+    const providersFromRegistry = yield call(getProvidersFromTDexRegistry);
```
Comment (altafan) [inline src/redux/sagas/tdexSaga.ts:66]: why did you remove the check for mainnet network? shouldn't this be done only in mainnet?

**Comment by tiero** [comment]
merged https://github.com/tdex-network/tdex-sdk/pull/118

File: src/redux/sagas/tdexSaga.ts (hunk 1 above)
Comment (louisinger) [reply src/redux/sagas/tdexSaga.ts]: U right, I've reverted this change

**Review by tiero — CHANGES_REQUESTED** [review]
Now, in case user edited (added/deleted) the provider list, we will always skip pulling from network. I think we should add a button in Settings > Manage.. page REFRESH to let user decide to pull the list from registry manually, ofc we should not replace/flush the manual added (just add eventually the one he deleted)

**Comment by tiero** [comment]
also sometimes it happens duplicated are added to the list, we should check an unique endpoint (host:port) is present in the list

**Comment by tiero** [comment]
duplicates are not enforced when adding new provider

image

File: src/pages/LiquidityProvider/index.tsx [hunk 2]
```diff
@@ -119,23 +125,28 @@ const LiquidityProviders: React.FC<LiquidityProvidersProps> = ({ providers }) =>
...
+                        color={isDuplicateProviderEndpoint() ? 'danger' : undefined}
                         required
                         value={newProviderEndpoint}
                         onIonChange={(e) => setNewProviderEndpoint(e.detail.value || '')}
                         inputmode="url"
                         placeholder="i.e http://localhost:9945"
                       />
+                      {isDuplicateProviderEndpoint() && (
+                        <span className="ion-text-right">
+                          <IonText color="danger">You already add this provider endpoint</IonText>
```
Comment (tiero) [inline src/pages/LiquidityProvider/index.tsx:137]:
```suggestion
                          <IonText color="danger">This provider already exists</IonText>
```

**Pull request merged**